File: scripts/seed.ts
import { config } from "dotenv";

config({ path: ".env.local" });
config({ path: ".env" });

import bcrypt from "bcryptjs";
import { neon } from "@neondatabase/serverless";
import { drizzle } from "drizzle-orm/neon-http";
import { eq } from "drizzle-orm";
import { DEFAULT_IPM_AREAS } from "../src/lib/real-checklist";
import { flatNamesToChecklistAreas } from "../src/lib/site-checklist";
import { SEED_RECORDS } from "../src/lib/seed-records";
import { clients, jobs, sites, users, visitRecords } from "../src/lib/db/schema";

const MANAGER_PASSWORD = "Password@123!";

async function main() {
  const url = process.env.DATABASE_URL;
  if (!url) {
    console.error("DATABASE_URL is not set. Add it to .env first.");
    process.exit(1);
  }

  const db = drizzle(neon(url));

  const existing = await db
    .select({ id: users.id })
    .from(users)
    .where(eq(users.email, "ops@qzone.co.ke"))
    .limit(1);

  if (existing.length > 0) {
    console.log("Seed skipped — data already present.");
    return;
  }

  const passwordHash = await bcrypt.hash(MANAGER_PASSWORD, 10);
  const today = new Date().toISOString().slice(0, 10);

  await db.insert(users).values([
    {
      id: "user-manager-ops",
      email: "ops@qzone.co.ke",
      name: "Operations Manager",
      passwordHash,
      role: "manager",
    },
    {
      id: "user-tech-boniface",
      email: "boniface@qzone.co.ke",
      name: "Boniface Kithinga",
      passwordHash,
      role: "technician",
    },
    {
      id: "user-tech-amina",
      email: "amina@qzone.co.ke",
      name: "Amina Wanjiru",
      passwordHash,
      role: "technician",
    },
  ]);

  await db.insert(clients).values({
    id: "client-kfc",
    name: "KFC",
    notes: "",
  });

  const seedChecklist = flatNamesToChecklistAreas([...DEFAULT_IPM_AREAS]);

  await db.insert(sites).values([
    {
      id: "site-01",
      clientId: "client-kfc",
      name: "Kakamega",
      address: "",
      areas: seedChecklist,
    },
    {
      id: "site-02",
      clientId: "client-kfc",
      name: "Westside Mall",
      address: "",
      areas: seedChecklist,
    },
  ]);

  await db.insert(jobs).values([
    {
      id: "visit-001",
      siteId: "site-01",
      technicianId: "user-tech-boniface",
      technicianName: "Boniface Kithinga",
      visitType: "full_inspection",
      date: today,
      status: "scheduled",
      timeWindow: "08:00 – 11:30",
    },
    {
      id: "visit-002",
      siteId: "site-01",
      technicianId: "user-tech-boniface",
      technicianName: "Boniface Kithinga",
      visitType: "follow_up",
      date: today,
      status: "scheduled",
      timeWindow: "12:00 – 13:00",
      followUpAreas: [
        "Fly Control Units (FCUs)",
        "Grease Trap",
        "Manholes & Drainage Systems",
      ],
      parentVisitId: "visit-001",
    },
    {
      id: "visit-003",
      siteId: "site-02",
      technicianId: "user-tech-amina",
      technicianName: "Amina Wanjiru",
      visitType: "full_inspection",
      date: today,
      status: "scheduled",
      timeWindow: "14:00 – 17:00",
    },
  ]);

  if (SEED_RECORDS.length > 0) {
    await db.insert(visitRecords).values(
      SEED_RECORDS.map((r) => ({
        id: r.id,
        jobId: r.visitId,
        siteId: r.siteId,
        clientName: r.clientName,
        siteName: r.siteName,
        visitType: r.visitType,
        technicianName: r.technicianName,
        date: r.date,
        submittedAt: new Date(r.submittedAt),
        areas: r.areas,
        reportText: r.reportText,
        submitId: `${r.visitId}:${r.date}`,
      })),
    );
  }

  console.log("Seed complete.");
  console.log(`Login: ops@qzone.co.ke / ${MANAGER_PASSWORD}`);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: api/scripts/seed.py
"""Seed pilot data from the existing Next.js prototype."""

from __future__ import annotations

import os
from datetime import date

os.environ.setdefault("JWT_SECRET", "dev-only-change-me-in-production")

from sqlalchemy import select

from app.core.security import hash_password
from app.db.session import SessionLocal
from app.models import Client, Job, JobStatus, Site, User, UserRole, VisitType

DEFAULT_IPM_AREAS = [
    "Red Dot Update",
    "Counter Section",
    "Soda Fridges",
    "Crusher & Ice Cream Machines",
    "Cash Counter",
    "Air Curtains",
    "Ceiling Section",
    "Under Seats",
    "Dispensers",
    "Washrooms",
    "Tumbling Machine & Marinator Machine",
    "DB Board",
    "Rounder & Burger Section",
    "Office Drawers & Shelves",
    "Trunking & Industrial Sockets",
    "Three-Compartment Sink",
    "Gaskets (Chillers & Freezers)",
    "Dry Goods Store",
    "Grease Trap",
    "Chips Upright Freezer",
    "Castor Wheels (Equipment)",
    "Staff Lockers",
    "Fly Control Units (FCUs)",
    "Non-Toxic Monitoring Devices",
    "Toxic Bait Stations",
    "Manholes & Drainage Systems",
]

MANAGER_PASSWORD = "Password@123!"


def seed() -> None:
    db = SessionLocal()
    try:
        if db.scalar(select(User).where(User.email == "ops@qzone.co.ke")):
            print("Seed skipped — data already present.")
            return

        manager = User(
            id="user-manager-ops",
            email="ops@qzone.co.ke",
            name="Operations Manager",
            password_hash=hash_password(MANAGER_PASSWORD),
            role=UserRole.manager,
        )
        tech_boniface = User(
            id="user-tech-boniface",
            email="boniface@qzone.co.ke",
            name="Boniface Kithinga",
            password_hash=hash_password(MANAGER_PASSWORD),
            role=UserRole.technician,
        )
        tech_amina = User(
            id="user-tech-amina",
            email="amina@qzone.co.ke",
            name="Amina Wanjiru",
            password_hash=hash_password(MANAGER_PASSWORD),
            role=UserRole.technician,
        )
        db.add_all([manager, tech_boniface, tech_amina])

        client = Client(id="client-kfc", name="KFC")
        db.add(client)

        site_kakamega = Site(
            id="site-01",
            client_id=client.id,
            name="Kakamega",
            areas=DEFAULT_IPM_AREAS,
        )
        site_westside = Site(
            id="site-02",
            client_id=client.id,
            name="Westside Mall",
            areas=DEFAULT_IPM_AREAS,
        )
        db.add_all([site_kakamega, site_westside])

        today = date.today().isoformat()
        jobs = [
            Job(
                id="visit-001",
                site_id=site_kakamega.id,
                technician_id=tech_boniface.id,
                technician_name=tech_boniface.name,
                visit_type=VisitType.full_inspection,
                date=today,
                status=JobStatus.scheduled,
                time_window="08:00 – 11:30",
            ),
            Job(
                id="visit-002",
                site_id=site_kakamega.id,
                technician_id=tech_boniface.id,
                technician_name=tech_boniface.name,
                visit_type=VisitType.follow_up,
                date=today,
                status=JobStatus.scheduled,
                time_window="12:00 – 13:00",
                follow_up_areas=[
                    "Fly Control Units (FCUs)",
                    "Grease Trap",
                    "Manholes & Drainage Systems",
                ],
                parent_visit_id="visit-001",
            ),
            Job(
                id="visit-003",
                site_id=site_westside.id,
                technician_id=tech_amina.id,
                technician_name=tech_amina.name,
                visit_type=VisitType.full_inspection,
                date=today,
                status=JobStatus.scheduled,
                time_window="14:00 – 17:00",
            ),
        ]
        db.add_all(jobs)
        db.commit()
        print("Seed complete.")
    finally:
        db.close()


if __name__ == "__main__":
    seed()
